math-scala/src/main/scala/org/apache/mahout/math/algorithms/regression/tests/FittnessTests.scala

```diff
@@ -47,10 +52,44 @@ object FittnessTests {
 
   // https://en.wikipedia.org/wiki/Mean_squared_error
   def MeanSquareError[R[K] <: RegressorModel[K], K](model: R[K], residuals: DrmLike[K]): R[K] = {
+    // TODO : I think mse denom should be (row - col) ??
     val mse = residuals.assign(SQUARE).sum / residuals.nrow
     model.mse = mse
     model.testResults += ('mse -> mse)
     model.summary += s"\nMean Squared Error: ${mse}"
     model
   }
+
+  // https://en.wikipedia.org/wiki/xxxx
+  def FTest[R[K] <: RegressorModel[K], K](model: R[K],  drmFeatures: DrmLike[K], drmTarget: DrmLike[K]): R[K] = {
+
+    // This is the residual sum of squares for just the intercept
+    //println(" drmTarget.ncol) = " +  drmTarget.ncol)
+    val interceptCol = drmTarget.ncol - 1
+    //val targetMean: Double = drmTarget
+    val targetMean: Double = drmTarget.colMeans().get(0)
+
+    val rssint: Double = ((drmTarget - targetMean  ).t %*% (drmTarget - targetMean)).zSum()
+    // ete above is the RSS for the calculated model
+
+    //println(" model.beta(0) = " +  model.beta(0))
+    //println(" model.beta(interceptCol) = " +  model.beta(interceptCol))
+    //println("rssint = " + rssint)
+    //println("rssmod = " + rssmod)
+
+    val groupDof = drmFeatures.ncol-1
+    val fScore = ((rssint - model.rss) / groupDof) / ( model.rss / (drmFeatures.nrow - groupDof- 1 ))
```

Comment:
this line is the crux of it- Looks good, but I'll feel better if you can verify with an R-prototype